lib/provider/cache.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Provider.Cache do
```

Comment:
Not insisting, but given the amount of code in this module, and the fact that each function is called exactly once from the single client module, I think this would have better worked as just inline `:ets` calls in the client module.

Reply:
That is very valid. I think I will just put those calls in the provider module.